### ISSUE TITLE
Allow "FormAlertWithSubmitButton" to handle more complex buttons

### ISSUE THREAD

**Issue by ctkochan22** [Expensify/App#9681]
## Problem:
We are using `FormAlertWithSubmitButton` in a lot of refactors for "Pattern C: Form blocking". However, there are a variety of buttons used in App. In my refactor alone, I have buttons with icons, `ButtonWithMenu`, and `MenuItem`. 

<img width="300" alt="image" src="https://user-images.githubusercontent.com/12980144/176976181-f0e9d79b-1d18-4ad3-ae17-eda2931e20c6.png">

<img width="300" alt="image" src="https://user-images.githubusercontent.com/12980144/176976188-7ff8b86d-6087-407d-b52c-c4ecb40ff6d0.png">

As stated here, we'd like the "offline" disabled version to mimic the original button as much as possible. And `FormAlertWithSubmitButton` only offers the generic `Button` component. So how can we use this component for more complex buttons.

## Solution:
Lets allow `FormAlertWithSubmitButton` to either take components as a prop, or allow the component to wrap custom buttons.

I have an example/draft PR here: https://github.com/Expensify/App/pull/9680

Wrapping would look like this:
```
            <FormAlertWithSubmitButton>
                    {isOffline => (
                        <Button
                            text={item.text}
                            icon={item.icon}
                            onPress={item.onPress}
                            isDisabled={item.isDisabled || isOffline}
                            style={item.style}
                            iconStyles={item.iconStyles}
                            success={item.success}
                            shouldShowRightIcon={item.shouldShowRightIcon}
                            extraLarge
                        />
                    )}
                </FormAlertWithSubmitButton>
```

**Comment by ctkochan22** [comment]
If people were curious, I originally add it as its [own component](https://github.com/Expensify/App/pull/9665) but thought it may be more efficient to merge it with `FormAlertWithSubmitButton`.

**Comment by ctkochan22** [comment]
Draft PR is out. Waiting on confirmation on what to do. Rory brings up a good point about whether it is strange that a button can also act as a wrapper: https://github.com/Expensify/App/pull/9680#issuecomment-1172788427

**Comment by ctkochan22** [comment]
After [discussing](https://expensify.slack.com/archives/C02HWMSMZEC/p1656717727334949) it with @tgolen @marcaaron @roryabraham , I updated this github's description as to the new plan.

We are going to go with a wrapper `FormAlertWrapper`. This wrapper will be used to add offline behavior and standardized error handling for more complex buttons that `FormAlertWithSubmitButton` cannot handle.
